src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_params.py

```diff
@@ -1005,3 +1023,147 @@ def _configure_security_policy_storage_params(arg_ctx):
+                   'Allowed values include: Gen4, Gen5.')
+
+        c.argument('storage_size_in_gb',
+                   options_list=['--storage'],
+                   arg_type=storage_param_type,
+                   help='Determines how much storage size to associate with Managed instance. '
+                   'Storage size must be specified in increments of 32 GB')
+
+        c.argument('license_type',
+                   arg_type=get_enum_type(DatabaseLicenseType),
+                   help='The license type to apply for this managed instance.')
+
+        c.argument('vcores',
+                   options_list=['--capacity', '-c'],
+                   help='Determines how much VCore to associate with Managed instance.')
+
+    with self.argument_context('sql managed-instance create') as c:
+        # Create args that will be used to build up the ManagedInstance object
+        create_args_for_complex_type(
+            c, 'parameters', ManagedInstance, [
+                'administrator_login',
+                'administrator_login_password',
+                'license_type',
+                'virtual_network_subnet_id',
+                'vcores',
+                'storage_size_in_gb'
+            ])
+
+        # Create args that will be used to build up the Managed Instance's Sku object
+        create_args_for_complex_type(
+            c, 'sku', Sku, [
+                'family',
+                'name',
+                'tier',
+            ])
+
+        c.ignore('name')  # Hide sku name
+
+        c.argument('administrator_login',
+                   options_list=['--admin-user', '-u'],
+                   required=True)
+
+        c.argument('administrator_login_password',
+                   options_list=['--admin-password', '-p'],
+                   required=True)
+
+        c.extra('vnet_name',
+                options_list=['--vnet-name'],
+                help='The virtual network name',
+                validator=validate_subnet)
+
+        c.argument('virtual_network_subnet_id',
+                   options_list=['--subnet'],
+                   required=True,
+                   help='Name or ID of the subnet that allows access to an Azure Sql Managed Instance. '
+                   'If subnet name is provided, --vnet-name must be provided.')
+
+        c.argument('assign_identity',
+                   options_list=['--assign-identity', '-i'],
+                   help='Generate and assign an Azure Active Directory Identity for this managed instance '
+                   'for use with key management services like Azure KeyVault.')
+
+    with self.argument_context('sql managed-instance update') as c:
+        # Create args that will be used to build up the ManagedInstance object
+        create_args_for_complex_type(
+            c, 'parameters', ManagedInstance, [
+                'administrator_login_password',
+            ])
+
+        c.argument('administrator_login_password',
```

Comment:
Instead of repeating `c.argument('administrator_login_password',` and `c.argument('assign_identity',`, you can just define them once in  in `with self.argument_context('sql managed-instance') as c:`,  section

Reply:
I am not actaully repeating them. They are different in properties. administrator_login_password is required when creating instance, and optional when updating it. And  assign_identity provides additional help info when doing update instance.

Reply:
Sounds good